core/src/main/java/org/mule/registry/SimpleRegistry.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.registry;
+
+import static org.reflections.ReflectionUtils.getAllFields;
+import static org.reflections.ReflectionUtils.withAnnotation;
+import org.mule.api.MuleContext;
+import org.mule.api.MuleException;
+import org.mule.api.lifecycle.Initialisable;
+import org.mule.api.lifecycle.InitialisationException;
+import org.mule.api.registry.LifecycleRegistry;
+import org.mule.api.registry.RegistrationException;
+import org.mule.lifecycle.phases.NotInLifecyclePhase;
+
+import java.lang.reflect.Field;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A very simple implementation of {@link LifecycleRegistry}. Useful for starting really lightweight
+ * contexts which don't depend on heavier object containers such as Spring or Guice (testing being
+ * the best example). <b>Do not use this registry for heavy duty production environments</b>.
+ * <p/>
+ * The {@link #inject(Object)} operation will only consider fields annotated with {@link Inject} and will perform
+ * the injection using simple, not-cached reflection. Again, not recommended for production environments.
+ *
+ * @since 3.7.0
+ */
+public class SimpleRegistry extends TransientRegistry implements LifecycleRegistry
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SimpleRegistry.class);
+    private static final String REGISTRY_ID = "org.mule.Registry.Simple";
```

Comment:
Yes I can. But the TransientRegistry was already like this so I figured "go with the flow"